dtfs-central-api/src/services/state-machines/fee-record/event-handlers/helpers/calculate-fixed-fee.test.ts

```diff
@@ -0,0 +1,193 @@
+import { ObjectId } from 'mongodb';
+import { when } from 'jest-when';
+import { addDays, addMonths, differenceInDays, startOfMonth, subMonths } from 'date-fns';
+import { Facility, ReportPeriod, TfmFacility } from '@ukef/dtfs2-common';
+import { calculateFixedFee } from './calculate-fixed-fee';
+import { TfmFacilitiesRepo } from '../../../../../repositories/tfm-facilities-repo';
+import { aFacility, aReportPeriod, aTfmFacility, aTfmFacilityAmendment } from '../../../../../../test-helpers/test-data';
+import { NotFoundError } from '../../../../../errors';
+
+console.error = jest.fn();
+
+describe('calculateFixedFeeAdjustment', () => {
+  const findOneByUkefFacilityIdSpy = jest.spyOn(TfmFacilitiesRepo, 'findOneByUkefFacilityId');
+
+  const facilityId = '12345678';
+  const tfmFacilityId = new ObjectId();
+
+  const utilisation = 100000;
+  const interestPercentage = 5;
+  const dayCountBasis = 365;
+
+  /**
+   * Date Details
+   *
+   * For the general test case, we want to set the report period
+   * start date to the start of this month, the cover end date to
+   * 365 days into the future and the cover start date to one month
+   * in the past. By doing this, we ensure that in the general case
+   * the calculation step where we use the value "days left in cover
+   * period" divided by `dayCountBasis` will always be 1, so it can
+   * be ignored.
+   */
+  const today = new Date();
+  const reportPeriodDate = startOfMonth(today);
+  const coverEndDate = addDays(reportPeriodDate, 365);
+  const coverStartDate = startOfMonth(subMonths(reportPeriodDate, 1));
+  const numberOfDaysRemainingInCoverPeriod = 365;
+
+  const reportPeriodStartMonth = reportPeriodDate.getMonth() + 1; // one-indexed
+  const reportPeriodStartYear = reportPeriodDate.getFullYear();
+  const reportPeriod: ReportPeriod = {
+    start: { month: reportPeriodStartMonth, year: reportPeriodStartYear },
+    end: { month: reportPeriodStartMonth, year: reportPeriodStartYear },
+  };
```

Comment:
These details are highly relevant to the calculation under test so any test with an assertion that relies on their specific value should have them specified directly in the test itself please